### ISSUE TITLE
Can create PRTB and CRTB with group and user targets

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#24036]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. create PRTB or CRTB with any combinations of groupId, groupPrincipalId, userId, userPrincipalId where at least two of them are not empty/null.

**Result:**
API returns successful, despite this being invalid and causing lifecycle to fail.

**Other details that may be helpful:**
Should be a validator that only allows one of the mentioned inputs.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
master 11/11/19

**Comment by jiaqiluo** [comment]
The bug fix is validated in rancher:master-head dad58367e

Run Rancher container and enable the Github Auth


**Do the following validations on  `v3/clusterRoleTemplateBindings`**
Validation 1: create a clusterRoleTemplateBindings with both userId and userPrincipalId
```
{
"clusterId": "c-rchvh",
"groupId": "",
"groupPrincipalId": "",
"name": "",
"namespaceId": "",
"roleTemplateId": "clusterroletemplatebindings-manage",
"userId": "u-z9lhp",
"userPrincipalId": "github_team://1395580"
}
```
Result: 
```
{
"baseType": "error",
"code": "InvalidBodyContent",
"message": "must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]",
"status": 422,
"type": "error"
}
```

Validation 2: create a clusterRoleTemplateBindings with both userId and groupId
```
{
"clusterId": "c-rchvh",
"groupId": "",
"groupPrincipalId": "github_team://1395580",
"name": "",
"namespaceId": "",
"roleTemplateId": "clusterroletemplatebindings-manage",
"userId": "u-z9lhp",
"userPrincipalId": ""
}
```
Result: 
```
{
"baseType": "error",
"code": "InvalidBodyContent",
"message": "must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]",
"status": 422,
"type": "error"
}
```
Validation 3: create a clusterRoleTemplateBindings without any id 
```
{
"clusterId": "c-rchvh",
"name": "",
"namespaceId": "",
"roleTemplateId": "cluster-owner"
}
```
Result: 
```
{
"baseType": "error",
"code": "InvalidBodyContent",
"message": "must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]",
"status": 422,
"type": "error"
}
```
Validation 4: create a clusterRoleTemplateBindings with proejct level roleTemplate
```
{
"clusterId": "c-rchvh",
"groupPrincipalId": "github_team://1395580",
"name": "",
"namespaceId": "",
"roleTemplateId": "project-member"
}
```
Result:
```
{
"baseType": "error",
"code": "InvalidBodyContent",
"message": "Cannot reference context [project] from [cluster] context",
"status": 422,
"type": "error"
}
```

Validation 5: create a clusterRoleTemplateBindings with one of those four fields
Results:
- creation the crtb is successful

**Comment by jiaqiluo** [comment]
The bug fix is validated in rancher:master-head f84c1e8f7

Run Rancher container and enable the Github Auth


**Do the following validations on  `v3/projectroletemplatebindings`**
Validation 1: create a projectroletemplatebindings with both userId and userPrincipalId
```
{
"groupId": "",
"groupPrincipalId": "",
"name": "",
"namespaceId": "",
"projectId": "c-fhg8x:p-wg6nn",
"roleTemplateId": "project-member",
"userId": "u-4cnrgx47ty",
"userPrincipalId": "github_org://9343010"
}

```
Result: 
- creation of the crtb is successful with the usage of the userPrincipalId


Validation 2: create a projectroletemplatebindings with both userId and groupPrincipalId
```
{
"groupId": "",
"groupPrincipalId": "github_team://1395580",
"name": "",
"namespaceId": "",
"projectId": "c-fhg8x:p-wg6nn",
"roleTemplateId": "project-member",
"userId": "u-4cnrgx47ty",
"userPrincipalId": ""
}
```
Result: 
```
{
"baseType": "error",
"code": "InvalidBodyContent",
"message": "must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]",
"status": 422,
"type": "error"
}
```
 
Validation 3: create a projectroletemplatebindings without any id 
```
 {
"groupId": "",
"groupPrincipalId": null,
"name": "",
"namespaceId": "",
"projectId": "c-fhg8x:p-wg6nn",
"roleTemplateId": "project-member",
"userId": null,
"userPrincipalId": ""
}
```
Result: 
```
 {
"baseType": "error",
"code": "InvalidBodyContent",
"message": "must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]",
"status": 422,
"type": "error"
}
```

Validation 4: create a projectroletemplatebindings with cluster-level roleTemplate
```
 {
"groupId": "",
"groupPrincipalId": "",
"name": "",
"namespaceId": "",
"projectId": "c-fhg8x:p-wg6nn",
"roleTemplateId": "cluster-member",
"userId": "u-4cnrgx47ty",
"userPrincipalId": ""
}
```
Result:
```
 {
"baseType": "error",
"code": "InvalidBodyContent",
"message": "Cannot reference context [cluster] from [project] context",
"status": 422,
"type": "error"
}
```

Validation 5: create a projectroletemplatebindings with one of those four fields
Results:
- creation of the prtb is successful
